src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java

```diff
@@ -0,0 +1,328 @@
+	 *
+	 * @param modID the id of the mod you want a simple pack for.
+	 */
+	public ConcreteResourcePack(String modID) {
+		super(getPackFileByModID(modID));
+		this.modID = modID;
+		this.cache = Maps.newHashMap();
+
+		// Obtain the real resource pack generated by forge.
+
+		IResourcePack realPack = FMLClientHandler.instance().getResourcePackFor(modID);
+		if (realPack instanceof LegacyV2Adapter) {
+			this.realResourcePack = (AbstractResourcePack) legacyPack.get(realPack);
+		} else if (realPack instanceof AbstractResourcePack) {
+			this.realResourcePack = (AbstractResourcePack) realPack;
+		}
+
+		if (realPack == null || this.realResourcePack == null) {
+			throw new MissingRealpackException(modID);
+		}
+
+		//Add our pack as a default pack, use FMLClientHandler's field for this so we don't need to worry about obf names
+		resourcePackList.get(FMLClientHandler.instance()).add(this);
+
+		// Confirms that our resourcepack is available as soon as possible to prevent missing resource errors.
+		if (Minecraft.getMinecraft().getResourceManager() instanceof SimpleReloadableResourceManager) {
+			SimpleReloadableResourceManager resourceManager = (SimpleReloadableResourceManager) Minecraft.getMinecraft().getResourceManager();
+			resourceManager.registerReloadListener(this);
+
+			// Forces this resource pack to be loaded early on. FML already did it's initial registration so we need to bypass that.
+			FallbackResourceManager domainManager = domainResourceManagers.get(resourceManager).get(modID);
+			resourcePacks.get(domainManager).add(this);
+		}
+	}
+
+	private static File getPackFileByModID(String modID) {
+		IResourcePack pack = FMLClientHandler.instance().getResourcePackFor(modID);
+		if (pack instanceof LegacyV2Adapter) {
+			return resourcePackFile.get(legacyPack.get(pack));
+		} else if (pack instanceof AbstractResourcePack) {
+			return resourcePackFile.get(pack);
+		}
+		return null;
+	}
+
+	public static ResourceLocation nameToLocation(String name) {
+		name = name.substring(name.indexOf("/") + 1);
+		String domain = name.substring(0, name.indexOf("/"));
+		String path = name.substring(name.indexOf("/") + 1);
+
+		LOG.debug("Converted " + name + " to " + new ResourceLocation(domain, path));
+		return new ResourceLocation(domain, path);
+	}
+
+	private static Map<Pair<RegistryDelegate<Item>, Integer>, ModelResourceLocation> getCustomModels() {
+		try {
+			Field field = ModelLoader.class.getDeclaredField("customModels");
+			return (Map<Pair<RegistryDelegate<Item>, Integer>, ModelResourceLocation>) FieldUtils.readStaticField(field, true);
+		} catch (Exception e) {
+			LOG.error("Caught exception getting customModels from the model loader, ", e);
+		}
+
+		return Maps.newHashMap();
+	}
+
+	@Override
+	protected InputStream getInputStreamByName(String name) throws IOException {
+		// Default to fallback if possible.
+		if (!((boolean) hasResourceName.invoke(realResourcePack, name))) {
```

Comment:
This should be covered by where in the resource pack list the resgen pack is, not by hacks like this. Make sure the order is such that the mod's real resources are preferred.